tools/omero/omero_filter.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import csv
+import json
+
+import ezomero as ez
+
+
+def filter_ids_ezo(user, pws, host, port, filter, id, value1, value2=None, tsv_file="filter_list.tsv"):
+
+    # Transform the id input in a list of integer
+    id = id.split(',')
+    id = list(map(int, id))
+
+    # Function to write tabular file from the ezomero output
+    def write_ids_to_tsv(data, header):
+        with open(tsv_file, 'a+', newline='') as f:
+            f.seek(0)
+            is_empty = f.tell() == 0  # Check if file is empty
+            writer = csv.writer(f, delimiter='\t')
+            if is_empty:
+                writer.writerow([header])  # Write the header
+            for item in data:
+                writer.writerow([item])  # Write each ID
+
+    try:
+        with ez.connect(user, pws, "", host, port, secure=True) as conn:
+            if filter == "filename":
+                fn_ids = ez.filter_by_filename(conn, id, value1)
+                write_ids_to_tsv(fn_ids, [f"Images with filename {value1}"])
+                return fn_ids
+            elif filter == "KP":
+                kp_ims = ez.filter_by_kv(conn, id, value1, value2)
+                write_ids_to_tsv(kp_ims, [f"Images with KV {value1} and {value2}"])
+                return kp_ims
+            elif filter == "tag":
+                tg_dict = ez.filter_by_tag_value(conn, id, value1)
+                write_ids_to_tsv(tg_dict, [f"Images with tag {value1}"])
+                return tg_dict
+
+            else:
+                raise ValueError(f"Unsupported object type")
+
+    except Exception as e:
```

Comment:
Is there a more specific `Exception` that we can catch in case of connection errors? Or should we just wrap the `filter_xyz()` calls in `try`-`except` blocks? The later would be my suggestion -- plus: do not just print the error (to stdout) but call `sys.exit(f"Connection error: {str(e)}")`

Also this would catch the `ValueErrors` that you raise here.